File: backend/app/models/execution.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base

class Execution(Base):
    __tablename__ = "executions"
    
    id = Column(Integer, primary_key=True, index=True)
    process_id = Column(Integer, ForeignKey("processes.id"))
    status = Column(String(50), nullable=False)  # 'running', 'completed', 'failed', 'stopped'
    output_path = Column(String(500))
    console_log = Column(Text)
    started_at = Column(DateTime(timezone=True), server_default=func.now())
    completed_at = Column(DateTime(timezone=True))
    
    # Relationship
    process = relationship("Process", back_populates="executions") 

File: frontend/types/execution.ts
export interface Execution {
  id: number;
  process_id: number;
  status: 'running' | 'completed' | 'failed' | 'stopped' | 'pending';
  output_path?: string;
  console_log?: string;
  started_at: string;
  completed_at?: string;
  process?: {
    id: number;
    name: string;
    description?: string;
    process_type: string;
  };
}

export interface ExecutionFilters {
  skip?: number;
  limit?: number;
}

export interface ExecutionUpdate {
  status?: string;
  output_path?: string;
  console_log?: string;
  completed_at?: string;
}

export interface ExecutionRequest {
  variables: Record<string, string>;
} 